### DISCUSSION TITLE
How to move Prisma Client in custom location to the dist folder when building for production?

### DISCUSSION THREAD

**Discussion by uncvrd** [prisma/prisma#10992] [Q&A] [answered]
Hi all! I have a specific use case where I need to generate the prisma client in a custom location outside of my `node_modules`, so I followed the documentation mentioned [here](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/generating-prisma-client#the-location-of-prisma-client) and have a `generator` that looks like the following:

```
generator prisma_client {
    provider        = "prisma-client-js"
    output          = "../generated/client"
    binaryTargets   = ["native", "linux-arm64-openssl-1.1.x"]
}
```

This obviously comes with the caveat where I need to import `PrismaClient` from an absolute path in my files like so:

```
import { PrismaClient } from "../../generated/client" 
```

This works fine while developing but I discovered that when I build my project for production, the `generated` directory does not get copied over to my `dist` folder that is configured as my `outputDir` in the `tsconfig`. As a result, the transpiled `.js` files in `dist` that reference this absolute path no longer can find the `PrismaClient`. 

I was wondering if there was a preferred method to handling how to move over the PrismaClient when building for production?

Thanks!

If you're curious here's a discussion why I need the custom location for the `PrismaClient` a few months ago: https://github.com/prisma/prisma/discussions/10280

**Comment by austincrim** [comment, marked as answer]
Hey @uncvrd 👋 ,

Could you copy the generated folder over in a `prebuild` script?

**Reply by uncvrd** [reply]
Thanks for the suggestion, yea I ended up having to do this process for now:

https://vccolombo.github.io/blog/tsc-how-to-copy-non-typescript-files-when-building/

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!

**Comment by bonarja** [comment]
`allowJs` and add files in include

tsconfig.json
```json
{
  "compilerOptions": {
    "outDir": "dist",
    "sourceMap": true,
    "allowJs": true
  },
  "include": ["src/**/*.ts", "src/generated/**/*"],
}

```
